stardict.cc

```diff
@@ -466,22 +466,22 @@ string StardictDictionary::handleResource( char type, char const * resource, siz
     {
       QString articleText = QString( "<div class=\"sdct_h\">" ) + QString::fromUtf8( resource, size ) + "</div>";
 
+      const QString srcRegexpStr(
+          "(<\\s*(?:img|script)\\s[^>]*src\\s*=)(?!\\s*[\"']?(?:data|https?|ftp):)(\\s*[\"']?)" );
```

Comment:
One more question, is `<img src= http://example.com/a.png>` valid HTML? (that is, with a space after `=` and without quotes around the URL). If not, should matching be prevented somehow?

Reply:
Now there is another thought. Why there is a negative lookup only for http/https/ftp/data protocols? If it will be a `gopher://` or let us say a `tftp://` link, well, it possibly will not work anyway, but is it a reason to prepend a `bres://` scheme?

May be it should be a negative lookup for `://`?

Reply:
> May be it should be a negative lookup for `://`?

Do you know where precisely GoldenDict should and should not insert `bres://`? I don't know for sure, but `bres` links are processed in *mainwindow.cc*, *articleview.cc* and *article_netmgr.cc* to extract resources from a dictionary file. Therefore I suppose `bres://` is needed or at least acceptable for whatever can be handled by code in those files and is potentially harmful whenever a web browser can follow the unaltered link.

Reply:
> extract resources from a dictionary file

Can a resource has e.g. `bres://{id}/file://filename.suf` form? It is a valid but malformed URL and also contains invalid file path (as for POSIX).

I am sure the substitution is not a security issues solver, because it handles only some specific cases excluding any form of `http://` links.

Reply:
You can test to answer your own questions. For StarDict, check what `resourceName` ends up in `StardictResourceRequest::run()` for different tricky URLs and figure out whether this `run()` function can possibly handle such a name.